chroma-manager/tests/integration/utils/chroma_log_collector.py

```diff
@@ -120,67 +142,97 @@ def fetch_log(self, server, source_log_path, destination_log_filename):
 
         # Generate the diagnostics from the server
-        result = shell_run(['ssh', server, 'chroma-diagnostics', '-v', '-v', '-v'])
+        result = shell_run(
+            ['ssh', server, 'iml-diagnostics', '--all-logs'], timeout=600)
 
         if result.timeout:
-            return["Chroma Diagnostics timed-out"]
+            return ["IML Diagnostics timed-out"]
 
-        # Find the diagnostics filename from the chroma-diagnostics output
-        cd_out = result.stderr.decode('utf8')
-        match = re.compile('/var/log/(diagnostics_.*\.tar\..*)').search(cd_out)
+        # Find the diagnostics filename from the iml-diagnostics output
+        cd_out = result.stdout.decode('utf8')
+        match = re.compile('/var/tmp/(sosreport-.*\.tar\..*)').search(cd_out)
         if not match:
-            return ["Did not find diagnostics filepath in chroma-diagnostics output:\nstderr:\n%s\nstdout:\n%s" %
-                    (cd_out, result.stdout.decode('utf8'))]
+            return [
+                "Did not find diagnostics filepath in iml-diagnostics output:\nstderr:\n%s\nstdout:\n%s"
+                % (cd_out, result.stdout.decode('utf8'))
+            ]
         diagnostics = match.group(1).strip()
 
         errors = []
 
-        # Copy and expand the diagnostics locally, so they will be able to be read in browser in Jenkins.
-        errors.append(self.fetch_log(server, "/var/log/%s" % diagnostics, ''))
-        errors.append(self.fetch_log(server, "chroma-diagnostics.log", '%s-chroma-diagnostics.log' % server))
-
-        if diagnostics.endswith('tar.lzma'):
-            if shell_run(['tar', '--lzma', '-xvf', "%s/%s" % (self.destination_path, diagnostics),
-                               '-C', self.destination_path]).rc:
-                errors.append("Error tar --lzma the chroma diagnostics file")
+        # Copy and expand the diagnostics locally, so they will be
+        # able to be read in browser in Jenkins.
+        errors.append(self.fetch_log(server, "/var/tmp/%s" % diagnostics, ''))
+
+        if diagnostics.endswith('tar.xz'):
+            if shell_run([
+                    'tar', '-xvJf',
+                    "%s/%s" % (self.destination_path, diagnostics), '-C',
+                    self.destination_path
+            ]).rc:
+                errors.append("Error tar --xvJf the iml diagnostics file")
         elif diagnostics.endswith('tar.gz'):
-            if shell_run(['tar', '-xvzf', "%s/%s" % (self.destination_path, diagnostics),
-                               '-C', self.destination_path]).rc:
-                errors.append("Error tar -xvzf the chroma diagnostics file")
+            if shell_run([
+                    'tar', '-xvzf',
+                    "%s/%s" % (self.destination_path, diagnostics), '-C',
+                    self.destination_path
+            ]).rc:
+                errors.append("Error tar -xvzf the iml diagnostics file")
```

Comment:
Would be nice to DRY out the above code by simply setting the tar decompresison switch in the if and applying that to a single `shell_run()` with something like:

```
if diagnostics.endswith('tar.xz'):
￼  sw = "J"
elif diagnostics.endswith('tar.gz'):
￼  sw = "z"
...
if shell_run([
￼        'tar', '-xv%sf',
        "%s/%s" % (sw, self.destination_path, diagnostics), '-C',
        self.destination_path
]).rc:
    errors.append("Error tar -xvzf the iml diagnostics file")
```